DocumentsFromSnapshotMigration/src/test/java/com/rfs/PerformanceVerificationTest.java

```diff
@@ -0,0 +1,133 @@
+package com.rfs;
+
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.store.ByteBuffersDirectory;
+import org.apache.lucene.util.BytesRef;
+import org.junit.jupiter.api.Test;
+
+import org.opensearch.migrations.reindexer.tracing.IDocumentMigrationContexts;
+
+import com.rfs.common.DocumentReindexer;
+import com.rfs.common.LuceneDocumentsReader;
+import com.rfs.common.OpenSearchClient;
+import com.rfs.tracing.IRfsContexts;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+public class PerformanceVerificationTest {
```

Comment:
Tuning this test might be hard - it will also operate differently on your machine, mine, github, and jenkins.  I'm happy to merge this change without a 'wall clock' based perf test.  What do you think about adding a disabled annotation and iterating on it in a future PR?

Reply:
Updated to a polling mechanism to not be dependent on the processing speed